library/src/main/java/com/schibsted/spain/barista/rule/BaristaRule.java

```diff
@@ -24,21 +25,26 @@
   private final ClearPreferencesRule clearPreferencesRule;
   private final ClearDatabaseRule clearDatabaseRule;
   private final ClearFilesRule clearFilesRule;
-  private final FlakyActivityTestRule<T> activityTestRule;
+  private final FlakyTestRule flakyTestRule;
+  private final ActivityTestRule<T> activityTestRule;
 
   private BaristaRule(Class<T> activityClass) {
     this.clearDatabaseRule = new ClearDatabaseRule();
     this.clearPreferencesRule = new ClearPreferencesRule();
     this.clearFilesRule = new ClearFilesRule();
-    this.activityTestRule = new FlakyActivityTestRule<>(activityClass,
-        INITIAL_TOUCH_MODE_ENABLED,
-        LAUNCH_ACTIVITY_AUTOMATICALLY)
+    this.flakyTestRule = new FlakyTestRule()
         .allowFlakyAttemptsByDefault(DEFAULT_FLAKY_ATTEMPTS);
+    this.activityTestRule = new ActivityTestRule<>(activityClass,
+        INITIAL_TOUCH_MODE_ENABLED,
+        LAUNCH_ACTIVITY_AUTOMATICALLY);
   }
 
   @Override
   public Statement apply(Statement base, Description description) {
-    return RuleChain.outerRule(activityTestRule)
+    return RuleChain.outerRule(flakyTestRule)
+        // ↓ All rules below flakyTestRule will be repeated
+        .around(activityTestRule)
+        // ↓ All rules below activityTestRule will execute before launching the activity
```

Comment:
Nice comments! 💘

Reply:
They're for future-me. Today I had to make some drawings to remember how RuleChain order works 😅

Reply:
I did the same yesterday 😢